src/utils/work/generatePlotWorkBody.js

```diff
@@ -0,0 +1,26 @@
+import plotNames from 'utils/plots/plotNames';
+
+const composeDotPlotWorkBody = (config) => ({
+  name: plotNames.workRequestName.DOT_PLOT,
+  markerGenes: !config.useCustomGenes,
+  input: {
+    nGenes: config.nMarkerGenes,
+    genes: config.selectedGenes,
+  },
+  subset: {
+    cellClassKey: config.selectedCellSet,
+    cellSetKey: config.selectedPoints,
+  },
+});
+
+const generatePlotWorkBody = (plotType, config) => {
+  switch (plotType) {
+    case plotNames.plotType.DOT_PLOT:
+      return composeDotPlotWorkBody(config);
+    default: {
```

Comment:
Yes. Other plots calculate plot data from data that is already available on the platform (e.g. gene expression), so they do not need to define their own work request body. This is the first plot that has its own work request to specifically calculate data for the plot. I think this functionality will be needed for new plots in the platform.

Reply:
in this situation it probably makes sense as I can foresee having a new plot that needs worker data. However, be very wary of coding for features not needed because it just adds extra complexity that might not be needed or used in the end. https://en.wikipedia.org/wiki/KISS_principle